### PR TITLE
Partially Fixes #3 : Added image and video subpage

### LINKED ISSUE #3: Create sub pages with different filters
The screenshot below shows different facettes/filters.

screenshot from 2016-10-15 15 44 05

Please implement it in the same style for
- [ ] All https://github.com/fossasia/loklak_search/issues/17
- [ ] Video Results https://github.com/fossasia/loklak_search/issues/49
- [ ] Image Results https://github.com/fossasia/loklak_search/issues/48
- [ ] News Results https://github.com/fossasia/loklak_search/issues/46

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Tabs  :- Tabs are used to switch from one media type sub page to other. I have used angular material for tabs. Every tabs gives out a value corresponding to which request is sent . For example, for tab with image, request is sent with query "query /image" and same for videos.
- Images :- Images with source as `"https://pbs.twimg.com/media/{{hashcode}}.jpg"` are only displayed because images on twitter have links just like this. All other images are not removed with source link as `http://pic.twitter.com/` because they are not images links.(need to be corrected from server side)  Currently, Images are displayed in a table format for multi photos depending on number of photos linked to tweets. Dimensions of images from server is large so it is changed so as to adjust to the feed -card. Multiple images are currently displayed in a table format just like this :-
 screen shot 2017-04-15 at 10 56 25 pm.
- Videos :- Use of iframe to display the videos. Links to videos are sanitized and bypassed Security Trust Resource. This will avoid us from XXS error. Moreover, In view directive is used to decrease the load time of video.

**Possible Issues**
- Top bar :- Top bar height has increased. This was done so as to adjust tabs in the top header bar.
- Logic for media specific tweets :- The logic is currently tied to a specific component. It should be a redux state which can be tapped upon by different components. Also, for specific media types, URL needs to be configured and effects needs to be checked so that it can be used later as a link. @hemantjadon @uttu357  @rishiraj824 please help with this. :)
- Lightbox :- UI/UX of lightbox needs to be worked upon. Since now media is also introduced to the lightbox, design needs to be adjusted so that no overflow occurs.
- Videos :- Loading time of videos is somewhat slow. So we can just show the thumbnail of the video and then load the video in the modal, like twitter does, this can also solve the problem of the site loading slowly.

**Link to live demo (if appropriate):** 

[Live Demo](https://achint08.github.io/loklak_search/)

### CODE REVIEW

**Comment by mariobehling** [comment]
Check out this design and sizes, please.
screenshot from 2017-04-18 18-11-06

**Comment by Achint08** [comment]
Hello @mariobehling , I will work on the design thing and the photo thing.
And regarding the sidebar and pop up thing, #326 will solve the issue most probably. It happens many a times that one of the variable goes undefined during processing of aggregations and other components of the web app crashes.

**Comment by Achint08** [comment]
Hello @mariobehling . Checkout now. (Y) 
[Gh-pages](https://achint08.github.io/loklak_search/)

File: src/app/feed/feed-card/feed-card.component.html [hunk 1]
```diff
@@ -30,6 +30,24 @@
 					[unshorten]="feedItem.unshorten || {}"
 					(click)="showLightBox.emit({show : 'hide'})"></feed-linker>
 			</div>
+			<br>
+			<table *ngIf="feedItem.images_count != 0">
```
Comment (hemantjadon) [inline src/app/feed/feed-card/feed-card.component.html]: Use flexboxes as a better design practice.

File: src/app/feed/feed-card/feed-card.component.ts [hunk 2]
```diff
@@ -126,4 +132,28 @@ export class FeedCardComponent implements OnInit {
...
+			let videoid = links[i].match(/(?:https?:\/{2})?(?:w{3}\.)?youtu(?:be)?\.(?:com|be)(?:\/watch\?v=|\/)([^\s&]+)/);
+			if(videoid != null) {
+   				links[i] = "http://www.youtube.com/embed/" + videoid[1];
+			}
+			links[i] = this.sanitizer.bypassSecurityTrustResourceUrl(links[i]);
+		})
+	}
+
+		private imageURLs(links) {
+		let img1 = new RegExp('https:\\/\\/abs\\.twimg\\.com\\/');
```
Comment (hemantjadon) [inline src/app/feed/feed-card/feed-card.component.ts]: Why regexp comparision is done?? I mean why cant we directly use `feeditem.images`

Comment (Achint08) [inline src/app/feed/feed-card/feed-card.component.ts]: Some of the image links are like `https://pic.twitter.com/{{hashcode}}.jpg` which are not the image links.
Moreover , some of them which are like `https://abs.twimg.com/{{hashcode}}.jpg` ...these images are the links for the emoticons used in the screen names.

Comment (hemantjadon) [inline src/app/feed/feed-card/feed-card.component.ts]: `https://pic.twitter.com/{{hashcode}}.jpg` Are these due to the scraping error from server?

Comment (Achint08) [inline src/app/feed/feed-card/feed-card.component.ts]: Yes.

File: src/app/feed/feed-card/feed-card.component.html (hunk 1 above)
Comment (Achint08) [inline src/app/feed/feed-card/feed-card.component.html]: Done. :)

**Comment by Achint08** [comment]
Any reviews @mariobehling @hemantjadon ? :)

**Comment by mariobehling** [comment]
Are you testing your work? E.g. most tweets in the video filter don't work.

Sorry, but it feels like this: You just re-submit this again and again without testing it yourself until it gets accepted.

**Comment by Achint08** [comment]
@mariobehling Sorry. I have no intention for that. I have used idea for requesting queries with videos from the server side. It seems sometimes for popular searches, we receive videos, For example , https://achint08.github.io/loklak_search/search?query=%23modi . Sometimes for non popular queries we are not getting it.  Guide me with this.

**Comment by mariobehling** [comment]
@Achint08 If this is a server issue please open an issue in the -> loklak server repository (https://github.com/loklak/loklak_server/) and report it and mention it in your pull request. Please help to make it easier for reviewers to understand what is going on and speed up the process.

**Comment by Achint08** [comment]
Opened issue loklak/loklak_server#1112 regarding media specific search.

**Comment by mariobehling** [comment]
Ok, thanks for opening the issue. Let's test this on live page and iterate more from here. Good job!

Please keep future commits smaller and easier to review. Cheers.

**Comment by mariobehling** [comment]
@Achint08 Please help to solve this issue on the server: https://github.com/loklak/loklak_server/issues/1112

**Pull request merged**